internal/controller/plugin/image_mirror_test.go

```diff
@@ -0,0 +1,148 @@
+		Expect(postRenderer).To(BeNil())
+	})
+
+	It("should create transformation preserving full image path", func() {
+		manifest := `image: ghcr.io/cloudoperators/greenhouse:main`
+		postRenderer := createRegistryMirrorPostRenderer(mirrorConfig, manifest)
+		Expect(postRenderer).NotTo(BeNil())
+		Expect(postRenderer.Kustomize.Images).To(HaveLen(1))
+		Expect(postRenderer.Kustomize.Images[0].Name).To(Equal("ghcr.io/cloudoperators/greenhouse"))
+		Expect(postRenderer.Kustomize.Images[0].NewName).To(Equal("mirror.example.com/ghcr-mirror/cloudoperators/greenhouse"))
+	})
+
+	It("should handle multiple images with different registries", func() {
+		manifest := `
+		containers:
+		- image: ghcr.io/cloudoperators/greenhouse:main
+		- image: docker.io/library/nginx:latest
+		- image: registry.k8s.io/pause:3.9
+		`
+		postRenderer := createRegistryMirrorPostRenderer(mirrorConfig, manifest)
+		Expect(postRenderer).NotTo(BeNil())
+		Expect(postRenderer.Kustomize.Images).To(HaveLen(2))
+
+		names := []string{postRenderer.Kustomize.Images[0].Name, postRenderer.Kustomize.Images[1].Name}
+		Expect(names).To(ConsistOf(
+			"ghcr.io/cloudoperators/greenhouse",
+			"docker.io/library/nginx",
+		))
+	})
+
+	It("should deduplicate identical images", func() {
+		manifest := `
+		containers:
+		- image: ghcr.io/cloudoperators/greenhouse:main
+		- image: ghcr.io/cloudoperators/greenhouse:main
+		- image: ghcr.io/cloudoperators/greenhouse:main
+		`
+		postRenderer := createRegistryMirrorPostRenderer(mirrorConfig, manifest)
+		Expect(postRenderer).NotTo(BeNil())
+		Expect(postRenderer.Kustomize.Images).To(HaveLen(1))
+		Expect(postRenderer.Kustomize.Images[0].Name).To(Equal("ghcr.io/cloudoperators/greenhouse"))
+		Expect(postRenderer.Kustomize.Images[0].NewName).To(Equal("mirror.example.com/ghcr-mirror/cloudoperators/greenhouse"))
+	})
+
+	It("should preserve nested repository paths", func() {
+		manifest := `image: ghcr.io/org/team/project/app:v1.0`
+		postRenderer := createRegistryMirrorPostRenderer(mirrorConfig, manifest)
+		Expect(postRenderer).NotTo(BeNil())
+		Expect(postRenderer.Kustomize.Images[0].Name).To(Equal("ghcr.io/org/team/project/app"))
+		Expect(postRenderer.Kustomize.Images[0].NewName).To(Equal("mirror.example.com/ghcr-mirror/org/team/project/app"))
+	})
+
+	It("should handle images with digest", func() {
+		manifest := `image: ghcr.io/cloudoperators/greenhouse@sha256:abc123`
+		postRenderer := createRegistryMirrorPostRenderer(mirrorConfig, manifest)
+		Expect(postRenderer).NotTo(BeNil())
+		Expect(postRenderer.Kustomize.Images[0].Name).To(Equal("ghcr.io/cloudoperators/greenhouse"))
+		Expect(postRenderer.Kustomize.Images[0].NewName).To(Equal("mirror.example.com/ghcr-mirror/cloudoperators/greenhouse"))
+	})
+
+	It("should handle images with quotes", func() {
+		manifest := `
+		containers:
+		- image: "ghcr.io/cloudoperators/greenhouse:main"
+		- image: 'docker.io/library/nginx:latest'
```

Comment:
`docker.io/library/nginx:latest` can also be specified as `nginx:latest`
Could you add a test case to ensure this handles the dockerhub default registry

[image registry defaulting](https://kubernetes.io/docs/concepts/containers/images/#:~:text=If%20you%20don%27t%20specify%20a%20registry%20hostname%2C%20Kubernetes%20assumes%20that%20you%20mean%20the%20Docker%20public%20registry.%20You%20can%20change%20this%20behavior%20by%20setting%20a%20default%20image%20registry%20in%20the%20container%20runtime%20configuration.)